File: app/src/main/java/mega/privacy/android/app/EphemeralCredentials.java
package mega.privacy.android.app;

public class EphemeralCredentials {
	private String email;
	private String password;
	private String session;
	private String firstName;
	private String lastName;

	public EphemeralCredentials(String email){
		this.email = email;
	}

	public EphemeralCredentials(String email, String password, String session, String firstName, String lastName) {
		this.email = email;
		this.password = password;
		this.session = session;
		this.firstName = firstName;
		this.lastName = lastName;
	}
	
	public String getEmail() {
		return email;
	}
	
	public void setEmail(String email){
		this.email = email;
	}

	public String getPassword() {
		return password;
	}

	public void setMyHandle(String password) {
		this.password= password;
	}
	
	public String getSession() {
		return session;
	}
	
	public void setSession(String session){
		this.session = session;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}
}

File: domain/src/main/kotlin/mega/privacy/android/domain/entity/login/EphemeralCredentials.kt
package mega.privacy.android.domain.entity.login

/**
 * Data class for storing ephemeral credentials when creating new account.
 *
 * @property email Account email.
 * @property password Account password. only available in the first time when creating an account, we don't persist it.
 * @property session Ephemeral session.
 * @property firstName Account first name.
 * @property lastName Account last name.
 */
data class EphemeralCredentials(
    val email: String?,
    val password: String? = null,
    val session: String?,
    val firstName: String?,
    val lastName: String?,
)
